src/main/java/ch/njol/skript/expressions/ExprNode.java

```diff
@@ -0,0 +1,142 @@
+ *
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package ch.njol.skript.expressions;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.classes.Changer.ChangeMode;
+import ch.njol.skript.classes.ClassInfo;
+import ch.njol.skript.config.Node;
+import ch.njol.skript.config.SectionNode;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.expressions.base.PropertyExpression;
+import ch.njol.skript.expressions.base.SimplePropertyExpression;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.Literal;
+import ch.njol.skript.lang.ParseContext;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.util.Kleenean;
+import ch.njol.util.coll.CollectionUtils;
+import org.bukkit.event.Event;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.*;
+
+@Name("Node")
+@Description({
+	"Returns a node inside a config (or another section-node).",
+})
+@Examples({})
+@Since("INSERT VERSION")
+public class ExprNode extends PropertyExpression<Node, Node> {
+
+	static {
+		Skript.registerExpression(ExprNode.class, Node.class, ExpressionType.PROPERTY,
+				"[the] node %string% (of|in) %node%",
+				"%node%'[s] node %string%",
+				"[the] nodes (of|in) %nodes%",
+				"%node%'[s] nodes"
+		);
+	}
+
+	private boolean isPath;
+	private Expression<String> pathExpression;
+
+	@Override
+	@SuppressWarnings("unchecked")
+	public boolean init(Expression<?>[] expressions, int pattern, Kleenean isDelayed, ParseResult parseResult) {
+		this.isPath = pattern < 2;
+		switch (pattern) {
+			case 0:
+				this.pathExpression = (Expression<String>) expressions[0];
+				this.setExpr((Expression<? extends Node>) expressions[1]);
+				break;
+			case 1:
+				this.pathExpression = (Expression<String>) expressions[1];
+				this.setExpr((Expression<? extends Node>) expressions[0]);
+				break;
+			default:
+				this.setExpr((Expression<? extends Node>) expressions[0]);
+		}
+		return true;
+	}
+
+	@Override
+	@SuppressWarnings("NullableProblems")
+	public Class<?> @Nullable [] acceptChange(ChangeMode mode) {
+		return null; // todo editable configs in future?
```

Comment:
Is there a reason why you're choosing to block changing the returned value? That is, if node has a default changer. It is typically better to leave this not overridden (if unused)

Reply:
These are almost always pure values (texts, numbers, maybe the odd UUID) so they probably shouldn't be 'changed' in any way. You should never really have something like a Player or whatever in a config that you'd actually want to change.

At this stage, configs aren't editable (that's in the follow up PR that we were still debating) so I'm not sure what changing the return value would actually mean.

I might be misunderstanding you, can you explain what you think would happen if you changed a return value here?

Reply:
Edit, I was reading the wrong file, this returns the node _itself_, not its value. Editing the node means editing the config structure, which isn't in the scope of this PR (this is read-only!)